nntrainer/models/neuralnet.cpp

```diff
@@ -629,6 +630,14 @@ int NeuralNetwork::train_run() {
 
   auto batch_size = std::get<props::TrainingBatchSize>(model_flex_props);
 
+  std::ofstream train_log_file;
+  auto &save_train_log_path =
+    std::get<props::SaveTrainLogPath>(model_flex_props);
+  if (!save_train_log_path.empty()) {
+    train_log_file =
+      checkedOpenStream<std::ofstream>(save_train_log_path, std::ios::out);
+  }
```

Comment:
consider adding an else case where `train_log_file` points to `/dev/null` (`train_log_file.open( "/dev/null", FLAGS);`).  Then you dont need to add checks on `train_log_file` or `save_train_log_path` property is valid or exists